frontend/www/js/omegaup/components/login/Signup.vue

```diff
@@ -192,6 +240,40 @@ export default class Signup extends Vue {
   expired(): void {
     this.recaptchaResponse = '';
   }
+
+  get maxDate() {
+    const currentYear = new Date().getFullYear();
+    return this.over13Checked
+      ? `${currentYear - 14}-12-31`
+      : `${currentYear}-12-31`;
+  }
+
+  get minDate() {
+    const currentYear = new Date().getFullYear();
+    return this.over13Checked ? '1900-01-01' : `${currentYear - 13}-01-01`;
+  }
+
+  checkAge() {
+    const dateOfBirth = new Date(this.dateOfBirth);
+    const today = new Date();
+    const age = today.getFullYear() - dateOfBirth.getFullYear();
+    this.isUnder13 = age > 13;
+    if (this.isUnder13) {
+      this.email = '';
+    }
+  }
+
+  updateDateRestriction() {
+    this.checkAge();
+
+    if (this.over13Checked) {
+      this.email = '';
+      this.isUnder13 = true;
+    } else {
+      this.parentEmail = '';
+      this.isUnder13 = false;
+    }
```

Comment:
```suggestion
    if (this.over13Checked) {
      this.email = '';
      this.isUnder13 = true;
      return;
    }
    this.parentEmail = '';
    this.isUnder13 = false;
```
Así nos evitamos el usar el `else`.